filechooser-portal/tests/PortalTester.vala

```diff
@@ -0,0 +1,100 @@
+public class PortalTester : Gtk.Application {
+    public PortalTester () {
+        Object (
+            application_id: "io.elementary.Files.PortalTester",
+            flags: ApplicationFlags.FLAGS_NONE
+        );
+    }
+
+    protected override void activate () {
+        var window = new Gtk.ApplicationWindow (this);
+        window.set_default_size (400, 400);
+        window.title = "Files Portal Tester";
+
+        var grid = new Gtk.Grid () {
+            halign = Gtk.Align.CENTER,
+            valign = Gtk.Align.CENTER,
+            orientation = Gtk.Orientation.VERTICAL,
+            row_spacing = 6,
+            margin = 6
+        };
+        var open_file_button = new Gtk.Button.with_label ("Open File");
+        var open_folder_button = new Gtk.Button.with_label ("Open Folder");
+        var save_button = new Gtk.Button.with_label ("Save");
+
+        grid.add (open_file_button);
+        grid.add (open_folder_button);
+        grid.add (save_button);
+
+        window.add (grid);
+
+        open_file_button.clicked.connect (() => {
+            var filechooser = new Gtk.FileChooserNative ("Custom Title", window, Gtk.FileChooserAction.OPEN, "Open", "Not Open");
+            filechooser.response.connect ((id) => {
+                if (id == Gtk.ResponseType.ACCEPT) {
+                    var message_dialog = new Gtk.MessageDialog (
+                        window,
+                        Gtk.DialogFlags.MODAL,
+                        Gtk.MessageType.INFO,
+                        Gtk.ButtonsType.CLOSE,
+                        "This file has been opened: %s",
+                        filechooser.get_file ().get_path ()
+                    );
+                    message_dialog.show_all ();
```

Comment:
Need to destroy dialog when closed

Reply:
Run is not necessary anymore (see https://developer.gnome.org/gtk4/unstable/ch41s02.html#id-1.7.4.4.77 for the reasoning behind its removal in GTK4)